### ISSUE TITLE
Reorganize the repo

### ISSUE THREAD

**Issue by ccharly** [ocsigen/ojwidgets#2]
Hi, to prepare the future release of ocsigen, we should sort all of the numerous modules of ojwidgets.

Indeed, as @Drup said in some commentaries, there are too many tools module. Should we create an Ojw_tools module and create modules inside ?

```
#Ojw_tools
Ojw_tools.Event(s)
Ojw_tools.Mobile
```

Or even better, maybe it's time to think about another tree structure for general application and mobile application ?

```
#Ojw_mobile
each module concerning mobile app ?
```

There is also too many modules which provides some helpers for JS in general, and they are splitted into modules and could be merge into an `Ojw_js` module ?

Also, bindings should be explicitly into an `Ojw_binding` module, don't you think ?

```
#Ojw_binding
Jcrop
Hammer_(s|S)wipe
```

About modules documentations, each module which is not enough documented, should be temporarily moved into a `experimental` directory ? Or do we let them like this in the repo ?

I'm going to sort all of them, but before doing that, I would like your opinion on it.

@jpdeplaix @Drup @balat @ArnaudParant

**Comment by Drup** [comment]
In any case, I think we can indeed put all utility functions in a `Ojw_tools` module, perhaps with some sub module for specific features (like `Touch` or `Events`). The module being designed not to be open, but used "as it" (`Ojw_tools.foo` directly). In general, add a submodule module `Foo` when you start to have several functions `foo_*".

For the rest, I think it's a good idea to go with `Ojw_foo` modules all over the place, one by widget. It follows the eliom convention and it will not pollute the namespace with generic names. It's a lot of module, but it's not an issue as long as we do the tools part correctly (and the documentation).
We should try to follow some conventions about those module, to have at least some coherence.

About isolating bindings in a specific module, I don't think it's a good idea to encode this in the module tree. This should be written in the documentation, by doing two lists of modules, one for bindings and one for pure ocaml code. Whatever a widget is a binding or not doesn't change the way you use it, so it makes no sense to call it differently.

About experimental widget, same thing, let's put them inside in the documentation, not in the hierarchy.

One big question remains about functors, and I'm not sure yet. @lisqlql we should have a direct discussion about that.

**Comment by ccharly** [comment]
You talk about `buttons` functors for example ?

If yes, first, I thought it was a good idea to do a 'bridge' between `Ew` and `Ojw`, but, maybe wrappers are the good way to do it. I mean, just thunk/jumper functions to do coercions and more. At least, for the documentations, it is better. But with functors, you don't need to do modifications in both libraries.

If not, which functors are you talking about ? :)

**Comment by Drup** [comment]
afair, functors for buttons and for scrollbar

**Comment by ccharly** [comment]
Finally I think that functors are really the best way of making widgets for `Ew` and `Ojw`. For maintainability purposes, and because you only need to write the code only once (okay, you need to provides coercion functions, but that's better than duplicate the code).

So I will stay with this method for now.
